### PR TITLE
feat(template): adding meson-python as a build-system option

### PR DESCRIPTION
**Pull request by YurelyCamacho**
## Pull Request description

- Activate option in TUI to meson-python.
- Adding meson-python as a build-system:
    - Creating a mesonpy-pyproject.toml
    - Editing post_gen_project.py
    - Adding mesonpy to build-system.sh
    - Adding meson.build as additional configuration file
    - Editing cookicutter.json
    - Adding meson-python in conda/dev.yaml 

Solve #55

## Pull Request checklists

This PR is a:
- [ ] bug-fix
- [x ] new feature
- [x ] maintenance

About this PR:
- [ ] it includes tests.
- [ ] the tests are executed on CI.
- [ ] the tests generate log file(s) (path).
- [ x] pre-commit hooks were executed locally.
- [ x] this PR requires a project documentation update.

Author's checklist:
- [ ] I have reviewed the changes and it contains no misspelling.
- [ ] The code is well commented, especially in the parts that contain more complexity.
- [ ] New and old tests passed locally.

## Reviewer's Checklist

- [ ] I managed to reproduce the problem locally from the `main` branch
- [ ] I managed to test the new changes locally
- [ ] I confirm that the issues mentioned were fixed/resolved .

### CODE REVIEW

**Comment by YurelyCamacho** [comment]
@xmnlab I have created this PR by adding the meson-python option as build system to see what happens with the tests. I still need to add the documentation, I will do it when I'm sure that everything else is working fine. Thanks

File: src/scicookie/{{cookiecutter.project_slug}}/conda/dev.yaml [hunk 1]
```diff
@@ -6,8 +6,10 @@ dependencies:
   - python <3.11
 {%- if cookiecutter.build_system == 'poetry' %}
   - poetry
-{%- else %}
+{%- elif cookiecutter.build_system == 'flit' %}
   - flit
+{%- else %}
```
Comment (xmnlab) [inline src/scicookie/{{cookiecutter.project_slug}}/conda/dev.yaml:11]:
```suggestion
{%- elif cookiecutter.build_system == 'mesonpy' %}
```

**Review by xmnlab — COMMENTED** [review]
@YurelyCamacho thanks for working on that. 
it is looking great.

just a small comment inline here

I hope to finish my PR fixing some general issues with the smoke test
and I will ping you to ask you to rebase your code.

**Review by xmnlab — COMMENTED** [review]
@YurelyCamacho looks good to me. 
could you rebase your branch on top of the upstream/main please?
just merged the PR fixing some issues with the smoke test.
also, as you mentioned on discord, could you ensure that these tests are being executed on CI please?
thanks!

**Comment by xmnlab** [comment]
@YurelyCamacho it seems your PR included changes from my last PR, and now it is blocking with some git conflicts.
let me know if you need any help to fix that

**Comment by xmnlab** [comment]
@YurelyCamacho in general looks good. could you add the build-system smoke to be executed on CI please?

**Comment by YurelyCamacho** [comment]
@xmnlab Didn't run build-system test, error with flit. What can I do now?
Another question: should I make any changes to the last PR that was accepted?

File: tests/smoke/base.sh [hunk 2]
```diff
@@ -44,6 +45,8 @@ if command -v poetry &> /dev/null; then
   poetry install
 elif command -v flit &> /dev/null; then
   flit install
+elif command -v mesonpy &> /dev/null; then
+  meson install
```
Comment (xmnlab) [inline tests/smoke/base.sh:49]: maybe the command would be `pip install .` instead

File: tests/smoke/base.sh [hunk 3]
```diff
@@ -44,6 +45,8 @@ if command -v poetry &> /dev/null; then
   poetry install
 elif command -v flit &> /dev/null; then
   flit install
+elif command -v mesonpy &> /dev/null; then
```
Comment (xmnlab) [inline tests/smoke/base.sh:48]: maybe it would be meson instead of meonspy

**Comment by YurelyCamacho** [comment]
@xmnlab It is a problem with the installation of the package. It could be something with the `pip install .` command. It could be `pip install -e .`.

**Comment by YurelyCamacho** [comment]
https://meson-python.readthedocs.io/en/latest/how-to-guides/editable-installs.html

**Comment by YurelyCamacho** [comment]
I was looking at this to see if it solved the error. https://mkdocstrings.github.io/python/usage/

**Comment by xmnlab** [comment]
@YurelyCamacho good job! it seems that worked! congrats :)
LGTM!

**Pull request merged**